### PR TITLE
VZA | Implement simplified assignment cancellation

### PR DESCRIPTION
**Pull request by mddilley**
Closes https://github.com/cityofaustin/atd-data-tech/issues/3614

I touched a lot of code here, but the changes boil down to:

1. Updating jQuery selectors to specifically target elements within a Knack view id
2. Abstracting the custom table so that `requestRecords()` accepts parameters that define:
a. which columns to display in the table and the functions that get the data from records to display in cells
b. whether to add the time filter buttons or not
c. the function of the "Cancel My Sign Up" button (can now serve the purpose of sign up and cancel OR cancel only)
d. what filters apply to the Knack request for the `officer_assignment` records that populate the tables

These changes allow existing code to be reused to add the custom table logic to the "My Assignments" and "Future Assignments" in the "My Assignments" view.

#### My Assignments view with no assignments today but future assignments
Screen Shot 2020-09-21 at 12 58 45 PM

#### My Assignments view with both today assignments and future assignments
Screen Shot 2020-09-21 at 1 02 20 PM

#### Cancellation modal that appears when cancelling from either table
Screen Shot 2020-09-18 at 3 34 20 PM

### CODE REVIEW

File: code/vza/vza.js [hunk 1]
```diff
@@ -754,57 +819,132 @@ $(document).on("knack-view-render.view_466", function (event, view, data) {
...
+    `;
+  }
+
+  return {
+    ...config,
+    columns: { Details: createFolderIconLink, ...config.columns }
+  };
+}
+
+$(document).on("knack-view-render.view_447", function (event, view, data) {
```
Comment (mateoclarke) [inline code/vza/vza.js:894]: did we want to use a config reference for this view?

File: code/vza/vza.js [hunk 2]
```diff
@@ -754,57 +819,132 @@ $(document).on("knack-view-render.view_466", function (event, view, data) {
...
+    },
+    addTimeFilters: false,
+    isCancelOnly: true
+  };
+
+  var updatedTableConfig = addRecordLinkToTableConfig(view, tableConfig);
+  var recordsTable = createRecordsTable(view.key, updatedTableConfig);
+
+  // Append table, then request records and append shifts to table body
+  $("#view_447" + "> div.view-header")
```
Comment (mateoclarke) [inline code/vza/vza.js:918]: same here

File: code/vza/vza.js [hunk 3]
```diff
@@ -754,57 +819,132 @@ $(document).on("knack-view-render.view_466", function (event, view, data) {
...
+    .after(recordsTable)
+    .ready(function () {
+      requestRecords(filters.today, view.key, updatedTableConfig);
+    });
+});
+
+$(document).on("knack-view-render.view_439", function (event, view, data) {
```
Comment (mateoclarke) [inline code/vza/vza.js:925]: same here

**Review by mateoclarke — APPROVED** [review]
🚢  

Looks really good and nice code commenting to help navigate. Caught a few hard coded view references but don't let that hold back from merging.

File: code/vza/vza.js (hunk 1 above)
Comment (mddilley) [reply code/vza/vza.js:897]: Yes, I will include these changes in my latest PR. 🙏

**Pull request merged**